react/components/BuyButton/index.js

```diff
@@ -64,7 +64,8 @@ export const BuyButton = ({
 }) => {
   const [isAddingToCart, setAddingToCart] = useState(false)
   const { showToast } = useContext(ToastContext)
-  const { settings: { promptOnCustomEvent }, showInstallPrompt } = usePWA()
+  const { settings = {}, showInstallPrompt  } = usePWA() || {}
+  const { promptOnCustomEvent = ""} = settings
```

Comment:
```suggestion
  const { promptOnCustomEvent } = settings
```

no need for be that defensive 😅